mysql-test/suite/rpl/t/rpl_bug-lp-1667307.test

```diff
@@ -0,0 +1,51 @@
+#
+# In the case if the bug is not fixed the test will crash on assert
+# in Rows_query_log_event::do_apply_event().
+#
+--source include/have_debug.inc
```

Comment:
For the release build the criteria would be admittedly weaker "runs, does not crash, completes correctly". According to your criteria the testcase should contain e.g. --source include/not_valgrind.inc, not_asan.inc at least, as it only trips on DBUG_ASSERT, shouldn't it? Having said that, if you considered everything above and still think this must be a debug-only testcase, then I'm OK with your decision. (I don't know about other reviewers of course)